### PR TITLE
Jetpack Cloud: improve our approach to fixing multiple threats

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

Things this PR fixes:
* Currently, when a user tries to fix all threats, a notice is displayed for every threat. Now this PR changes that and only one notice is displayed.
* Because fixing/ignoring a threat happens in the background and it can take any amount of time, sometimes a threat in the UI stays forever as disabled (in progress) until the user moves to another page or refresh the tab. This is how the process works now:

```
            t1                t2                       t3                t4
request fix/ignore ---> operation succeeded ---> request scan state ---> ????
```

It can happen that even in t4, the threat has not been fixed/ignored yet. If this happens, the UI stays the same until the users do something that triggers a new request for the state of the Scan. This PR fix that by pooling until we know all threats have been fixed.

#### Implementation notes

The new pooling mechanism works only if the following conditions are true:

* The action includes `pooling: true`
* The scan state is `idle`
* There are threats that are being fixed/ignored (`fixerStatus === 'in_progress'`)

This means, the app will keep asking `/scan` every `POOL_EVERY_MILLISECONDS` until all threats are fixed/ignored.

#### Testing instructions

* Go to http://jetpack.cloud.localhost:3000/
* Select a site that has Scan and at least two threats
* Go to Scan -> Scanner
* Click the `[Fix all]` button and confirm the action
* Verify only one notice is displayed
* Verify request are sent to `/scan` until all threats are fixed (no threat has the `fixer_status` property)
* Verify all threats are remove from the screen and you are back to the Scan is okay UI

Fixes 1151678672052943-as-1175373711120913

#### Demo

Kapture 2020-05-13 at 15 53 17

**If the state of the Scan looks like this, the app should keep pooling until all `fixer_status` properties disappear from the threats.**
image

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/scan/threats/fix-all-threats.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		),
+		...action.threatIds.map( ( threatId ) => updateThreat( action.siteId, threatId ) ),
+	];
+};
+
+// We don't have a wait to only execute this path if all request succeeded.
+export const success = ( action ) => [
+	successNotice(
+		i18n.translate(
+			"We're hard at work fixing these threats in the background. Please check back shortly."
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/scan/threats/fix-all-threats.js:48]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'We're hard at work fixing this threat in the background. Please check back shortly.' )` *ES Score: 8*

File: client/state/data-layer/wpcom/sites/scan/threats/fix-all-threats.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			duration: 4000,
+			id: action.noticeId,
+		}
+	),
+	requestScanStatus( action.siteId, true ),
+];
+
+// Not sure if this is even going to happen. Maybe if all of them fail.
+export const failure = ( action ) => [
+	errorNotice( i18n.translate( 'Error fixing threats. Please contact support.' ), {
```
Comment (a8ci18n) [inline client/state/data-layer/wpcom/sites/scan/threats/fix-all-threats.js:60]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Error fixing threat. Please contact support.' )` *ES Score: 8*

**Comment by mattgawarecki** [comment]
I tested and was able to see the new notifications and correct copy -- that part looks great. 👍 Pooling works as expected, and I only see one notification for all the threat fixers, as planned.

Not strictly relevant to this PR, but I did notice that while threats are being fixed, the only progress indication we get right now is from the alerts in the top right, which quickly disappear. As the threats are fixed, they simply "blip" out of existence, which looks a bit confusing as a user.

Do we want to ask for design input on a potential new "fixing" state? In my head this would be similar to a loading or scanning state, except with copy to indicate that threats are in the process of being fixed. This may actually supersede the need for an alert notification all together, if we did it this way.

**Comment by rcanepa** [comment]
> Not strictly relevant to this PR, but I did notice that while threats are being fixed, the only progress indication we get right now is from the alerts in the top right, which quickly disappear. As the threats are fixed, they simply "blip" out of existence, which looks a bit confusing as a user.
> 
> Do we want to ask for design input on a potential new "fixing" state? In my head this would be similar to a loading or scanning state, except with copy to indicate that threats are in the process of being fixed. This may actually supersede the need for an alert notification all together, if we did it this way.

I agree 100%. There is a lot of room for improvement in that regard. I also imagine something happening inside the Threat that's being fixed. Maybe a tiny progress bar or loading indicator that shows the user specifically which threats are being fixed.

**Pull request merged**